src/NuGet.Clients/NuGet.VisualStudio.Implementation/Extensibility/VsPackageInstaller.cs

```diff
@@ -515,11 +515,11 @@ private SourceRepository GetSource(string source)
             // install the package
             if (package.Version == null)
             {
-                await packageManager.InstallPackageAsync(nuGetProject, package.Id, resolution, projectContext, sources, Enumerable.Empty<SourceRepository>(), token);
+                await packageManager.InstallPackageAsync(nuGetProject, package.Id, resolution, projectContext, new SourceCacheContext(), sources, Enumerable.Empty<SourceRepository>(), token);
```

Comment:
If there are places with a lot of default `SourceCacheContext` instances (seems to be the vast majority), an overload would be appropriate.